### ISSUE TITLE
Settings QA: 

### ISSUE THREAD

**Issue by rishig** [zulip/zulip-desktop#566]
Copying from https://chat.zulip.org/#narrow/stream/16-desktop/topic/settings.20qa.
I expect almost all of these will be fast changes, so maybe as first step you can fix or disagree with all the easy ones (e.g. wording changes, quick style fixes), and then we can open new issue(s) with the remaining list. 

### Settings
- [ ] In the Settings menus, sometimes the back button is active and works, sometimes it is active and doesn't work, and sometimes it is not active. I wasn't able to get reliable reproducers, but if you click reload, different settings tabs, and the back button in random orders, maybe you can repro it.
- [x] The menu items on the left I think could have more vertical padding
- [x] The buttons could be nicer (e.g. rounded corners, on-hover behavior of Add and Choose is broken in the General tab)
- [ ] We don't really do directional shadows in Zulip, so I would consider making the on-hover of the settings sections have centered shadows

### Settings: General
- [x] Show sidebar setting -> either (Cmd/Ctrl+S) or (Cmd+S or Ctrl+S)
- [x] Desktop Notification -> Desktop Notifications
- [ ] Show Desktop Notifications -> Should be "Suppress desktop notifications" or "Block desktop notifications from Zulip" to match "Mute all sounds from Zulip" (that's also a more accurate description of the setting).
- [x] Spellchecker -> "spellchecker" or "spell checker"
- [x] Add custom CSS -> I'd put this in Advanced, just as "Add custom CSS" with a button "Upload stylesheet"
- [x] Show downloaded file in the file manager -> "Show downloaded files in file manager"
- [x] Default download location button: "Change" instead of "Choose",
- [x] Download location: maybe indent this?
- [x] The text gets too close to the button at certain sizes (see image on czo)

### Settings: Network
- [ ] (requires restart) -> it seems like it does the restart automatically?
- [ ] The manual proxy config also seems to do an automatic restart, and possibly Save whether or not you click Save
- [x] The checkmark in the Save button is confusing, I think.

### Settings: Organization
- [x] I think this could use a bit more horizontal spacing between the icon, realm name, and the open icon (see image on czo)
- [x] We don't really use : in these situations, and they can be removed (see image on czo)
- [x] Custom CA's certificate file -> Certificate file
- [x] Add -> Upload

### Settings: Shortcuts
- [x] The tip at the bottom should be at the top, and doesn't need to be in a Tip styling (i.e. just text is fine)

**Comment by akashnimare** [comment]
> In the Settings menus, sometimes the back button is active and works, sometimes it is active and doesn't work, and sometimes it is not active. I wasn't able to get reliable reproducers, but if you click reload, different settings tabs, and the back button in random orders, maybe you can repro it.

This one is the expected behaviour. It only gets activated when it could go backwards. Also, it's disabled for the Setting page.

> The buttons could be nicer (e.g. rounded corners, on-hover behavior of Add and Choose is broken in the General tab)

This is done in the latest release.

![image](https://user-images.githubusercontent.com/2263909/46000705-f453f280-c0c6-11e8-8e8a-c61485635c49.png)

**Comment by akashnimare** [comment]
> Show Desktop Notifications -> Should be "Suppress desktop notifications" or "Block desktop notifications from Zulip" to match "Mute all sounds from Zulip" (that's also a more accurate description of the setting).

I think **Show Desktop Notifications** makes sense here since by default it's turned on and if we choose to rename it like "Block desktop notifications from Zulip" we need to play with the default notification setting also which I don't is a good idea.

> (requires restart) -> it seems like it does the restart automatically?
  The manual proxy config also seems to do an automatic restart, and possibly Save whether or not you click Save

This is just to make sure the settings gets apply successfully, @abhigyank right?

> Download location: maybe indent this?

IMO, this looks good without the indentation.

**Comment by abhigyank** [comment]
> (requires restart) -> it seems like it does the restart automatically?

No, currently it doesn't. Need to quit the app and restart.

>  The manual proxy config also seems to do an automatic restart, and possibly Save whether or not you click Save

It doesn't do a restart. I suppose it does a hard reload on clicking save.

**Comment by akashnimare** [comment]
@rishig can you re-visit the setting page again and see if the changes are good enough? I think we have covered most of the points from the checklist.

**Comment by rishig** [comment]
cool, looks great! 

The remaining things that stand out:
* [ ]  Show Desktop Notifications -> Should be "Suppress desktop notifications" or "Block desktop notifications from Zulip" to match "Mute all sounds from Zulip" (that's also a more accurate description of the setting).
I think this one is pretty important, since from the current setting name I think most users would have an incorrect intuition as to what that setting does. 

* [ ] The vertical space here is still a bit confusing:
![image](https://user-images.githubusercontent.com/890911/48748470-e2f11780-ec2b-11e8-85db-8f18faca602c.png)

**Comment by akashnimare** [comment]
Yeah, we could change the notification wording but this should be done carefully since this could override the notification settings and complicate things. The changes will require -
* Rename the setting [here](https://github.com/zulip/zulip-electron/blob/master/app/renderer/js/pages/preference/general-section.js#L50)
* Rename `showNotifications` to maybe `blockNotifications`. Delete the old value from the db and replace it with the new name + it's value.
* As of now the default value for this is `true`, we need to change it to `false`
https://github.com/zulip/zulip-electron/blob/master/app/renderer/js/main.js#L111
https://github.com/zulip/zulip-electron/blob/master/app/renderer/js/main.js#L119

**Comment by akashnimare** [comment]
> Show Desktop Notifications -> Should be "Suppress desktop notifications" or "Block desktop notifications from Zulip" to match "Mute all sounds from Zulip" (that's also a more accurate description of the setting).

We could also rename the `Mute all sounds from Zulip` to `Play notification sound` or something so that it matches with the flow and it would be a bit less complicated I think.

**Comment by akashnimare** [comment]
> The vertical space here is still a bit confusing

@rishig does this look good?
![image](https://user-images.githubusercontent.com/2263909/48967604-1c999800-f009-11e8-8b43-9b3c81991416.png)

**Comment by kanishk98** [comment]
> (requires restart) -> it seems like it does the restart automatically?
> The manual proxy config also seems to do an automatic restart, and possibly Save whether or not you click Save

@rishig can you tell me if this is still an issue on your end? I just checked these out on my system and I seem to be getting expected behaviour. (manual restart in the first case, second case requiring us to click save before saving and then automatically reloading the app)

**Comment by rishig** [comment]
Not sure how this got lost -- sorry! 
* The vertical spacing for download location looks good
* hm, I'm not on master for the desktop app currently, but could you send a gif of what happens when you click on "use system proxy settings" twice?
